openhands/controller/agent_controller.py

```diff
@@ -223,6 +227,19 @@ async def on_event(self, event: Event):
         if not any(isinstance(event, filter_type) for filter_type in self.filter_out):
             self.state.history.append(event)
 
+        # Check if adding this event would exceed context window
+        if self.max_input_tokens is not None:
+            # Create temporary history with new event
+            temp_history = self.state.history + [event]
+            token_count = self.agent.llm.get_token_count(temp_history)
+
```

Comment:
I think here I've had an issue in the past, because litellm's token counter uses only a handful of possible tokenizers (those for very popular LLMs or something). So the token counting method looks good, but it doesn't always work. 😅 I think for many LLMs it gives an Exception - if I recall correctly.

On the bright side, their implementation accepts a custom tokenizer. 

So an alternative could be to define a custom tokenizer as a configuration setting, for users to set it to what they need, and we use it in our `get_token_count` method, something [like this](https://github.com/All-Hands-AI/OpenHands/pull/4510/files#diff-b46ec9dd4b319d75809520e34b2559fd2ce46ca4895709eae9312aa04a8fa9a4R382).